### DISCUSSION TITLE
allow to change Domain

### DISCUSSION THREAD

**Discussion by happyreacer** [nextcloud/all-in-one#362] [Ideas] [answered]
Hi,
is it possible to change my Domain? Or what is an alternative way? Possibly via a backup in a new setup?

**Comment by szaimen** [comment]
Hi, this is currently not possible and I am also not sure if it makes sense for the project as you are not able to change the domain in the android app and desktop client either, afaik (see https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/domain_change.html). Either way I'll leave it open for now but it will not be implemented in the near future.
Pull Requests are welcome!

**Reply by dkebler** [reply]
so if I used a temporary domain  e.g.  test.domain.com  and now want to switch to files.domain.com  that is not possible?   As it it I can point caddy to files.domain.com but then the domain is rewritten to test.domain.com.    Can't I go into the apache container maybe and change this>

**Reply by szaimen** [reply]
technically you could change the domain in the configuration.json...
Afterwards the containers need to be restarted from the aio interface...

**Reply by szaimen** [reply]
@dkebler did it work?

**Reply by szaimen** [reply]
Another way for you would be starting a new aio instance without any data...

**Reply by cerebrux** [reply]
> technically you could change the domain in the configuration.json... Afterwards the containers need to be restarted from the aio interface...

Worked like charm ! Here are the steps that I followed for a AIO setup that is behind Caddy proxy

1. In your AIO nextcloud, go to Settings -> Overview -> AIO Interface. There you should stop your containers and probably make a backup. Don't close your tab or browser with the AIO interface just yet.
2. Open your caddy file `sudo vim /etc/caddy/Caddy` and copy the configuration you have for `old.domain.com` and create below it a `new.domain.com` configuration. Save, and restart your caddy service `sudo systemectl restart caddy`
3. Edit the nextcloud AIO configuration file e.g.  `vim /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/configuration.json` by searching and changing every occurrence of the `old.domain.com` to `new.domain.com` . Save and exit your editor
4. Return to your AIO Interface tab on your browser and click `start containers`. Once they have started, you can visit `new.domain.com`. 
5. Open your caddy file `sudo vim /etc/caddy/Caddy`  and remove any configuration for `old.domain.com` then save, and restart your caddy service

**Reply by rTfact** [reply]
This worked for me:
1. Stop running containers via Web-interface (optionally backup)
2. Rename host
$ sudo hostnamectl set-hostname `new.domain.com`
3. Edit NextCloud configuration.json file - replace `old.domain.com` with `new.domain.com`
$ sudo nvim /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/configuration.json
4. reboot
$ sudo systemctl isolate reboot

**Reply by guupt** [reply]
> > technically you could change the domain in the configuration.json... Afterwards the containers need to be restarted from the aio interface...
> 
> Worked like charm ! Here are the steps that I followed for a AIO setup that is behind Caddy proxy
> 
>     1. In your AIO nextcloud, go to Settings -> Overview -> AIO Interface. There you should stop your containers and probably make a backup. Don't close your tab or browser with the AIO interface just yet.
> 
>     2. Open your caddy file `sudo vim /etc/caddy/Caddy` and copy the configuration you have for `old.domain.com` and create below it a `new.domain.com` configuration. Save, and restart your caddy service `sudo systemectl restart caddy`
> 
>     3. Edit the nextcloud AIO configuration file e.g.  `vim /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/configuration.json` by searching and changing every occurrence of the `old.domain.com` to `new.domain.com` . Save and exit your editor
> 
>     4. Return to your AIO Interface tab on your browser and click `start containers`. Once they have started, you can visit `new.domain.com`.
> 
>     5. Open your caddy file `sudo vim /etc/caddy/Caddy`  and remove any configuration for `old.domain.com` then save, and restart your caddy service

It also worked for me. Thank you.

**Comment by michaelso** [comment]
how about a change of internal ip? i did change my server's IP in the local network after setting up nextcoud-aio, and not sure how to change the local internal ip for accessing the aio panel

**Comment by szaimen** [comment]
> how about a change of internal ip? i did change my server's IP in the local network after setting up nextcoud-aio, and not sure how to change the local internal ip for accessing the aio panel

are you talking about the automatic login button that still has the old ip-address?

**Comment by michaelso** [comment]
no, it is in the https://domain.name/settings/admin/overview
[Open Nextcloud AIO Interface ↗](https://IP:8080/api/auth/getlogin?token=xxx)
Hope there is a way to update the IP in this case when the server is moved internally

**Comment by szaimen** [comment]
> no, it is in the https://domain.name/settings/admin/overview
> [Open Nextcloud AIO Interface ↗](https://IP:8080/api/auth/getlogin?token=xxx)

Yes, I thought so. 

You can do so by yourself by simply visiting the AIO interface from the new domain/ip-adress/port, click on stop containers and start containers -> done.

**Comment by epifanio** [comment]
Hi, attemting to do the same I am stuck at the following log - after container started with the new DNS:

```
listen.allowed_clients = 127.0.0.1,::1,172.18.0.14,172.18.0.2
+ set +x
[05-Jun-2024 03:23:40] NOTICE: fpm is running, pid 683
[05-Jun-2024 03:23:40] NOTICE: ready to handle connections
Activating Collabora config...
✓ Reset callback url autodetect
Checking configuration
🛈 Configured WOPI URL: https://next.new.dns/
🛈 Configured public WOPI URL: https://next.old.dns
🛈 Configured callback URL: 

Failed to fetch discovery endpoint from https://next.new.dns/
cURL error 7: Failed to connect to next.new.dns port 443 after 3067 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://next.new.dns/hosting/discovery
```

**Reply by epifanio** [reply]
from the apache logs i can see the following:

```

{"level":"error","ts":1717565239.509789,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"next.new.dns","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"0.0.0.0: Timeout during connect (likely firewall problem)","instance":"","subproblems":[]}}
{"level":"error","ts":1717565239.5103946,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"next.new.dns","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"0.0.0.0: Timeout during connect (likely firewall problem)","instance":"","subproblems":[]},"order":"https://acme-staging-v02.api.letsencrypt.org/acme/order/150587344/16988028314","attempt":1,"max_attempts":3}
{"level":"error","ts":1717565239.510605,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"next.new.dns","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 400 urn:ietf:params:acme:error:connection - 0.0.0.0: Timeout during connect (likely firewall problem)"}
{"level":"error","ts":1717565239.5107768,"logger":"tls.obtain","msg":"will retry","error":"[next.new.dns] Obtain: [next.new.dns] solving challenge: next.new.dns: [next.new.dns] authorization failed: HTTP 400 urn:ietf:params:acme:error:connection - 0.0.0.0: Timeout during connect (likely firewall problem) (ca=https://acme-staging-v02.api.letsencrypt.org/directory)","attempt":16,"retrying_in":1800,"elapsed":14613.504801333,"max_duration":2592000}

```

I replaced  an IP address with 0.0.0.0 - it wasn't my maxgine IP, so I guess something to do with my office network or even the letsencrypt  IP ?

**Reply by epifanio** [reply]
I fixed, it was an error on my side (non-matching IP:DNS) - the recipe provided here worked fine!

**Comment by philiprenich** [comment]
Throwing in my two cents.

This won't update any existing in-app links you have - say between pages within Collectives, etc. You'll need to update that stuff manually.

Also, there are a handful of settings within NC you'll need to update. I just logged in to my admin account and walked through all the admin settings to update the URL of any stray settings. It will depend on your config & apps but is easy to click through and find any old URLs and change them.

Also, just to make sure I didn't get locked out, I reset my 2FA since the code which the 2FA app uses to generate referenced the old URL. It still worked, but it made me nervous so I disabled and re-configured it (maybe better would be disable prior to starting the process)

**Comment by surapuramakhil** [comment]
# What about Automated or UI intuitive way for updating domains? does it always need to be manual process?

**Reply by szaimen** [reply]
> does it always need to be manual process?

Yes. See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-domain

**Reply by surapuramakhil** [reply]
thanks for sharing.

can we have feature request for this?

**Comment by szaimen** [comment, marked as answer]
Just to mention it: we have the following docs on the topic nowadays: https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-domain
